sql/src/main/java/io/crate/execution/engine/aggregation/GroupingProjector.java

```diff
@@ -68,16 +68,21 @@ public GroupingProjector(List<? extends Symbol> keys,
         if (keys.size() == 1) {
             Symbol key = keys.get(0);
             if (key.valueType().equals(DataTypes.LONG)) {
-                collector = new GroupBySingleLongCollector(
-                    collectExpressions,
-                    mode,
-                    functions,
-                    inputs,
-                    ramAccountingContext,
-                    (Input<Long>) keyInputs.get(0),
-                    indexVersionCreated,
-                    bigArrays
-                );
+                collector = createSingleNumberGroupingCollector(Long.class,
```

Comment:
Couldn't we pass in the `key.valueType()` and just have a single `numericTypes.contains(keyType)` check to avoid the `else if` branches?

Reply:
I don't understand this exactly. We'd still need to pass in the correct Java type right? And we need only a specific set of numeric types? 
Or are you thinking of having of map from crate DataType to Java Type?